tests/TypecheckSpec.hs

```diff
@@ -20,44 +21,55 @@ spec = do
   void $ runIO $ inferModule False "examples/accounts/accounts.repl" "accounts"
   checkFuns
 
-topLevelChecks :: Text -> (TopLevel Node,TcState) -> Spec
-topLevelChecks n (tl,s) = do
-  it (unpack n ++ " typechecks") $
-    filter isUnresolvedTy (map _aTy (toList tl)) `shouldBe` []
-  it (unpack n ++ " has no failures") $ toList (_tcFailures s) `shouldBe` []
+type TCResult = (TopLevel Node,TcState)
```

Comment:
Putting it in `Pact.Typechecker` makes sense to me.  But if we put it there, it seems like it might be better to make it `data` or `newtype`.  If we keep it here, having it as a type alias makes more sense to me.